financial-connections/src/main/java/com/stripe/android/financialconnections/domain/SaveAccountToLink.kt

```diff
@@ -19,64 +24,108 @@ internal class SaveAccountToLink @Inject constructor(
         phoneNumber: String,
-        selectedAccounts: Set<String>,
-        country: String
-    ): FinancialConnectionsSessionManifest = runCatching {
-        repository.postSaveAccountsToLink(
-            clientSecret = configuration.financialConnectionsSessionClientSecret,
-            email = email,
-            country = country,
-            phoneNumber = phoneNumber,
-            locale = (locale ?: Locale.getDefault()).toLanguageTag(),
-            consumerSessionClientSecret = null,
-            selectedAccounts = selectedAccounts
-        )
+        selectedAccounts: List<PartnerAccount>,
+        country: String,
+        shouldPollAccountNumbers: Boolean,
+    ): FinancialConnectionsSessionManifest {
+        return ensureReadyAccounts(shouldPollAccountNumbers, selectedAccounts) { selectedAccountIds ->
+            repository.postSaveAccountsToLink(
+                clientSecret = configuration.financialConnectionsSessionClientSecret,
+                email = email,
+                country = country,
+                phoneNumber = phoneNumber,
+                locale = (locale ?: Locale.getDefault()).toLanguageTag(),
+                consumerSessionClientSecret = null,
+                selectedAccounts = selectedAccountIds,
+            )
+        }
     }
-        .updateCustomSuccessMessage(selectedAccounts.size)
-        .getOrThrow()
 
     suspend fun existing(
         consumerSessionClientSecret: String,
-        selectedAccounts: Set<String>,
-    ): FinancialConnectionsSessionManifest = runCatching {
-        repository.postSaveAccountsToLink(
+        selectedAccounts: List<PartnerAccount>,
+        shouldPollAccountNumbers: Boolean,
+    ): FinancialConnectionsSessionManifest {
+        return ensureReadyAccounts(shouldPollAccountNumbers, selectedAccounts) { selectedAccountIds ->
+            repository.postSaveAccountsToLink(
+                clientSecret = configuration.financialConnectionsSessionClientSecret,
+                email = null,
+                country = null,
+                phoneNumber = null,
+                locale = null,
+                consumerSessionClientSecret = consumerSessionClientSecret,
+                selectedAccounts = selectedAccountIds,
+            )
+        }
+    }
+
+    private suspend fun ensureReadyAccounts(
+        shouldPollAccountNumbers: Boolean,
+        partnerAccounts: List<PartnerAccount>,
+        action: suspend (Set<String>) -> FinancialConnectionsSessionManifest,
+    ): FinancialConnectionsSessionManifest {
+        val selectedAccountIds = partnerAccounts.map { it.id }.toSet()
+        val linkedAccountIds = partnerAccounts.mapNotNull { it.linkedAccountId }.toSet()
+
+        val pollingResult = if (shouldPollAccountNumbers) {
+            pollAccountNumbers(linkedAccountIds)
+        } else {
+            Result.success(Unit)
+        }
+
+        return pollingResult.onFailure {
+            disableNetworking()
+        }.mapCatching {
```

Comment:
We disable networking if the polling fails.

@kgaidis-stripe The default polling logic is that we don’t retry on network errors, but only on 202 headers. Should we change that?

Reply:
My first instinct is that I believe network issues should _generally_ be handled by the networking/internet "infrastructure" as a whole (the networking library, the device, the router etc. lol)

Or, in other words, this particular polling purpose is to handle Stripe/FC-specific logic/polling

Reply:
Sure, but we don’t have a general retry mechanism for requests. This is something that has to happen in the domain layer, as it might not always be desirable.

I believe we should continue polling if we encounter a network issue, and only continue without networking if we don’t get a response within 5 second (250 ms interval * 20 attempts).